res/controllers/Pioneer-DDJ-SR2-scripts.js

```diff
@@ -561,40 +552,21 @@ DDJSR2.MixerContainer = function() {
         group: "[Master]",
         inKey: "crossfader",
     });
-    DDJSR2.panels = [false, false];
-    this.panelSelectButton = function(channel, control, value, _status, _group) {
-        if (value) {
-            if ((DDJSR2.panels[0] === false) && (DDJSR2.panels[1] === false)) {
-                DDJSR2.panels[0] = true;
-            } else if ((DDJSR2.panels[0] === true) && (DDJSR2.panels[1] === false)) {
-                DDJSR2.panels[1] = true;
-            } else if ((DDJSR2.panels[0] === true) && (DDJSR2.panels[1] === true)) {
-                DDJSR2.panels[0] = false;
-            } else if ((DDJSR2.panels[0] === false) && (DDJSR2.panels[1] === true)) {
-                DDJSR2.panels[1] = false;
-            }
 
-            engine.setValue("[Samplers]", "show_samplers", DDJSR2.panels[0]);
-            engine.setValue("[EffectRack1]", "show", DDJSR2.panels[1]);
-        }
-    };
+    const panelStates = new DDJSR2.RingBufferView([[false, false], [true, false], [true, true], [false, true]]);
 
-    this.shiftPanelSelectButton = function(channel, control, value, _status, _group) {
+    const makePanelSelectButton = method => function(channel, control, value, _status, _group) {
         if (value) {
-            if ((DDJSR2.panels[0] === false) && (DDJSR2.panels[1] === false)) {
-                DDJSR2.panels[1] = true;
-            } else if ((DDJSR2.panels[1] === true) && (DDJSR2.panels[0] === false)) {
-                DDJSR2.panels[0] = true;
-            } else if ((DDJSR2.panels[0] === true) && (DDJSR2.panels[1] === true)) {
-                DDJSR2.panels[1] = false;
-            } else if ((DDJSR2.panels[1] === false) && (DDJSR2.panels[0] === true)) {
-                DDJSR2.panels[0] = false;
-            }
+            const [showSamplers, showFX] = method(); // Call method as a function
 
-            engine.setValue("[Samplers]", "show_samplers", DDJSR2.panels[0]);
-            engine.setValue("[EffectRack1]", "show", DDJSR2.panels[1]);
+            engine.setValue("[Samplers]", "show_samplers", showSamplers);
+            engine.setValue("[EffectRack1]", "show", showFX);
         }
     };
+
+    // Pass the bound method reference
+    this.panelSelectButton = makePanelSelectButton(panelStates.next.bind(panelStates));
+    this.shiftPanelSelectButton = makePanelSelectButton(panelStates.previous.bind(panelStates));
```

Comment:
using bind like this is a little ugly.
```suggestion
    this.panelSelectButton = makePanelSelectButton(() => panelStates.next());
    this.shiftPanelSelectButton = makePanelSelectButton(() => panelStates.previous());
```